### ISSUE TITLE
Advanced Planet Configuration Problems

### ISSUE THREAD

**Issue by Tarkov** [zmaster587/AdvancedRocketry#903]
Minecraft Version:
-1.7.10
Forge Version:
-forge10.13.4.161
LibVulpes Version:
-1.7.10-0.2.6
Advaced Rocketry Version:
-1.7.10-1.2.5
*is this the latest curseforge build?
Yes

When I use my own planetDefs.xml the stars and planets are mixed together, some are missing and everything is a mess.

I tried my template with the Dim ID and Dim Mapping attributes, also without them, with other mods, and just with Advanced Rocketry and Libvulpes. I never have the stars, planets and satellites organised as I want them. Also it is being impossible to set up the moon in the overworld.

What can I do?

**Comment by gdarai** [comment]
You can post here your cfg file and we will look at it.

**Comment by Tarkov** [comment]
PlanetDefs: [https://pastebin.com/2Dijxx0w](url)
Config file: [https://pastebin.com/2gL3P9X5](url)

Thanks!

**Comment by gdarai** [comment]
Ok, I checked your planetDefs.xls file and there are two reasons why your space system is a mess and a simple solution.

Reason 1) That file is a mess:
Explicitly: You have there multiple solar systems with a lot of planets very close to each other. All of them has no theta and phi 0. Several of them has orbital distance very close to each other like 100 and 99 (Earth, Mercury, Venus) Some of them are in wrong places, like Moon should probably be orbitting Earth, not Sun, right?

Reason 2) Advanced Rocketry is not finished and has still some significant bugs. Like: Theta and Phi are messed up and does not work the way it is described on the page.

Solution:
You need to start from scratch and move in little steps. I don't know what 1.7 can and cannot do, so read the mod's page carefully ```http://65.78.115.159:8000/AdvancedRocketry/config/AdvancedPlanetConfiguration.php``` Create new instance of minecraft with only advanced rocketry so you can restart the game quickly. Start by having the overworld and add one of the planets and develop on that. Add next and next, tweak them one by one.
```
<galaxy>
    <star name="Sol" temp="100" x="0" y="0" size="1.0" numPlanets="0" numGasGiants="0">
        <planet name="Earth" DIMID="0" dimMapping="">
            <isKnown>true</isKnown>
            <fogColor>1.0,1.0,1.0</fogColor>
            <skyColor>1.0,1.0,1.0</skyColor>
            <gravitationalMultiplier>100</gravitationalMultiplier>
            <orbitalDistance>98</orbitalDistance>
            <orbitalPhi>0</orbitalPhi>
            <rotationalPeriod>24000</rotationalPeriod>
            <atmosphereDensity>100</atmosphereDensity>
        </planet>
        <planet name="Mercury" DIMID="-5" dimMapping="">
            <isKnown>true</isKnown>
            <fogColor>1.0,1.0,1.0</fogColor>
            <skyColor>1.0,1.0,1.0</skyColor>
            <gravitationalMultiplier>37</gravitationalMultiplier>
            <orbitalDistance>100</orbitalDistance>
            <orbitalPhi>0</orbitalPhi>
            <rotationalPeriod>24000</rotationalPeriod>
            <atmosphereDensity>0</atmosphereDensity>
        </planet>
     </star>
</galaxy>
```
Even in this example you can check the functionality of phi and orbital distance. Also try different rotational perides and warping from planet to planet.

Beware issues with Planet Selectors and other visual bugs described here on this Issue tracker.

**Comment by dmodoomsirius** [comment]
http://arwiki.dmodoomsirius.me/AdvancedRocketry/config/AdvancedPlanetConfiguration.php

**Comment by Tarkov** [comment]
Thank you gdarai and dmodoomsirius for your answers. I think I did not explained it clearly When I said that the planets were mixed together I meant that some planets went to different stars and others were missing, changing everything. I do not mind the visual of the planet selector.

I followed the config guide very carefully from the beginning. And did not make any mistake in my files. Everything is planned, the moon it is orbiting the sun because I cannot make it orbit the earth, and the planets are in very close orbital distance because I was trying to make it as realistic as possible, they are proportionally located like in the the Solar System. It may looked like a mess, but a very organised mess.

But after many tries I think the problem it is when the planets have a very similar name, like my Trappist system, when they are called "Trappist b","Tappist c"..

Thank you anyway.
